checker/src/main/java/org/checkerframework/checker/nullness/NullnessNoInitVisitor.java

```diff
@@ -811,7 +814,26 @@ public Void visitDoWhileLoop(DoWhileLoopTree tree, Void p) {
     @Override
     public Void visitConditionalExpression(ConditionalExpressionTree tree, Void p) {
         checkForNullability(tree.getCondition(), CONDITION_NULLABLE);
-        return super.visitConditionalExpression(tree, p);
+        // Note: A copy of condThen is provided as an argument to the second invocation of
+        // NullnessVisitor#commonAssignmentCheck. This approach is taken due to the method's side
+        // effects now.
+        // A more appropriate handling of the conditional expression would involve removing
+        // replacePolyQualifier from commonAssignmentCheck, and then managing the two branches
+        // separately within Nullnesstransfer.
+        // This check's logic should be kept in sync with the super method.
+        AnnotatedTypeMirror condThen = atypeFactory.getAnnotatedType(tree);
+        AnnotatedTypeMirror condElse = condThen.deepCopy();
+        this.commonAssignmentCheck(
+                condThen, tree.getTrueExpression(), "conditional.type.incompatible");
+        this.commonAssignmentCheck(
+                condElse, tree.getFalseExpression(), "conditional.type.incompatible");
+        // Avoid calling super, as it will trigger a false negative in the false branch of super
```

Comment:
`trigger a false negative` sounds odd, as for a false negative nothing happens. What did you mean?

Reply:
I will try to say the super method does not handle conditional branches correctly and we use a hack here.

Reply:
I don't think anything was wrong with the super method... if so, we should fix it there or at least also add a comment there.

Reply:
how about `Avoid calling super, as the current method uses a hack and does not sync with the super method. Instead, manually implement the logic from TreeScanner#visitConditionalExpression to traverse the subtree.`